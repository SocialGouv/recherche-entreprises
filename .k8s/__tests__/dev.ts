//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/github-actions.env";

jest.setTimeout(1000 * 60);
test("kosko generate --dev", async () => {
  expect(
    await getEnvManifests("dev", "", {
      ...project("recherche-entreprises").dev,
      RANCHER_PROJECT_ID: "c-bd7z2:p-wbfnt",
    })
  ).toMatchSnapshot();
});
