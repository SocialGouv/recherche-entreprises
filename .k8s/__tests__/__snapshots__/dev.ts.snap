// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --dev 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    socialgouv/creator: autodevops
    janitor/ttl: 7d
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: c-bd7z2:p-7ms8p
    git/branch: refs/heads/mybranch
    git/remote: socialgouv/recherche-entreprises
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/recherche-entreprises
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    azure-pg-admin-user: recherche-entreprises
    application: recherche-entreprises
    component: recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  name: recherche-entreprises-mybranch
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/recherche-entreprises
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: recherche-entreprises-api
    application: recherche-entreprises
    component: recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  name: recherche-entreprises-api
  namespace: recherche-entreprises-mybranch
spec:
  replicas: 1
  selector:
    matchLabels:
      app: recherche-entreprises-api
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/recherche-entreprises
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: recherche-entreprises-api
        application: recherche-entreprises
        component: recherche-entreprises
        owner: recherche-entreprises
        team: recherche-entreprises
        cert: wildcard
    spec:
      containers:
        - image: harbor.fabrique.social.gouv.fr/cdtn/recherche-entreprises-api:1.5.8
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: recherche-entreprises-api
          ports:
            - containerPort: 3000
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 16Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          envFrom:
            - secretRef:
                name: elastic-recherche-entreprises-read
---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  name: elastic-recherche-entreprises-read
  annotations:
    sealedsecrets.bitnami.com/cluster-wide: 'true'
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/recherche-entreprises
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    application: recherche-entreprises
    component: recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  namespace: recherche-entreprises-mybranch
spec:
  encryptedData:
    ELASTICSEARCH_URL: >-
      AgCxK1MqqVkCOHvMjjcieUsj1uE/ATUzAfYTYVEMqcWFxTnugOaYlXB2mTuZtzdjXNvINREzDynnVRBghPIJvrJm59vPgexC5FyQptdPf2XwQAG6n0yMLWAEBM+O8ieO1D6vBu6BYuq4QYoItq+SMTE6APNX+p6wi4/LP04cR8p8bVKNxckPMfyVfQGhUSTD+2gC5ow2x1Mx67odqqP3E8eK9tm5tU6ltzOpKqJKITUQIqTQbSKFr8p77DE4IrdbPn8MpvdP+VrT+ztF4thPkpVpuYWy9rABdQKeEuNdN2AuvmPeJs/2B80iApIgtWWQjV/iG+WlLGMohvWDZVsBGuBuYGzhjJlr5w8OljwKDihb2cG1M/Ot/LM1xRuCC85wvqH0oMhnnP/3DZts5aYAN+qjAezOkPT9kjT+gu0WoQB+E5r9ITcYYHYbJw/YSxS2z86Ul0amRRW4ovAWT5c/nFDZU6i74I8YyVh8oOtrieZgNF4ajJcIr5YpjT2LICKoZR0KExrQ4V30zZJ+wZcvd8W3e/KCmKsFxloTkLq/sjEiFQoe6ffYTQhSUXZynqrdUKZsdoSleprfctlJtf3rzxt5hZIpcLulXrYk7pPPiHkTcV3Zc4zkN3BFE1A9vsgTvh6xTXlvUekJHHMXU3t8ZyZD8vvemxv0FE8G33XWQF2DEkmyc3gXMlfPDLho4fHv0h0iId0c9klFkOpmp8REFWJyuRQjq+hOHvfoCSu1ymsCwA0CMAguJoX1XEhn0aA9QIgf0VyBe/cVJwu/GfW74UBi6x5feqzFovYIUXbDP6kW
    ELASTICSEARCH_API_KEY: >-
      AgCtqcSVac6ptXOzSGqrBps1Uz6elTOnheaAf8cqAb1G960NCqmmkx1L1FDhfSKCuhBUAp0mT/3rokX3cg/pLIqImIVlPEhq/n+ZaKPHqGt0UT9QpD3r1C4nYan/WJ1PnBlCTXYlA/dtmCakNjtf2ikvWg/6Keu1OGlHt/ymVc2guxqGl++caxPiEIQAiVHDZksc09RM2Z6jU/WnI5gzTPQjgucb2yiHBLWZsEcoCIsA2nXrp2/vQ/ATMtITKb1223jyHnJiargNnQrRVRA7Geuz5FyaqSXjqvjG1HrqVxAlVNzFPb/N3fN8C6KHCjtOuu/pOM+cqC4cFvzyzqGymcDOKGC5Ax47j5v0KsVlJZ93dfO2yD/Fi2txeuNdmQ42bvhpGbPJu/iYLXmaETvu6/qpL3UrU4i2KVyxXW0CbUzBVh1fTmCClAMDgETpWNvnRXgAuu0jbgTnWjN0zSmjia0/BJmY1Z5uct7T4Lli4Dz/vJucBthIBVgYaVX/jLqfaNWpD1TuWkTHsf4gR5HErzQp60NDej/I7TespFMqRMhrroc6VaH/Wl6wV4msRM3sYaxcHPd4kJxrOeA7cQnfqQsWs4/RG++p+TCmD18qgevVBV1z11vFu2R6KE+c1YRANE5e6EBMO8yUly1yarrx6jOo6E5kAdKqPGkNUyq3BgOvGWmkbV32slQ4hXm7zUuyVnXBmpmoryowHhqV7iW8Si6CSMKACCENVqJPrk55VaL8sw8dZfQF0aekJoEqf/UqAv/DZUTYOAg4QI+4iD4=
  template:
    metadata:
      annotations:
        sealedsecrets.bitnami.com/cluster-wide: 'true'
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/recherche-entreprises
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      name: elastic-recherche-entreprises-read
      labels:
        application: recherche-entreprises
        component: recherche-entreprises
        owner: recherche-entreprises
        team: recherche-entreprises
        cert: wildcard
    type: Opaque
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: recherche-entreprises-api
    application: recherche-entreprises
    component: recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  name: recherche-entreprises-api
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/recherche-entreprises
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: recherche-entreprises-mybranch
spec:
  ports:
    - name: http
      port: 80
      targetPort: 3000
  selector:
    app: recherche-entreprises-api
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/recherche-entreprises
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: recherche-entreprises-api
    application: recherche-entreprises
    component: recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  name: recherche-entreprises-api
  namespace: recherche-entreprises-mybranch
spec:
  rules:
    - host: api-recherche-entreprises-mybranch.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: recherche-entreprises-api
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - api-recherche-entreprises-mybranch.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/recherche-entreprises
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: recherche-entreprises-search
    application: recherche-entreprises
    component: recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  name: recherche-entreprises-search
  namespace: recherche-entreprises-mybranch
spec:
  replicas: 1
  selector:
    matchLabels:
      app: recherche-entreprises-search
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/recherche-entreprises
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: recherche-entreprises-search
        application: recherche-entreprises
        component: recherche-entreprises
        owner: recherche-entreprises
        team: recherche-entreprises
        cert: wildcard
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/recherche-entreprises/search:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: recherche-entreprises-search
          ports:
            - containerPort: 3000
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 16Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          envFrom:
            - secretRef:
                name: elastic-recherche-entreprises-read
          env:
            - name: ELASTICSEARCH_INDEX_NAME
              value: search-entreprises
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: recherche-entreprises-search
    application: recherche-entreprises
    component: recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  name: recherche-entreprises-search
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/recherche-entreprises
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: recherche-entreprises-mybranch
spec:
  ports:
    - name: http
      port: 80
      targetPort: 3000
  selector:
    app: recherche-entreprises-search
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/recherche-entreprises
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: recherche-entreprises-search
    application: recherche-entreprises
    component: recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  name: recherche-entreprises-search
  namespace: recherche-entreprises-mybranch
spec:
  rules:
    - host: search-recherche-entreprises-mybranch.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: recherche-entreprises-search
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - search-recherche-entreprises-mybranch.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/recherche-entreprises
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: recherche-entreprises-front
    application: recherche-entreprises
    component: recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  name: recherche-entreprises-front
  namespace: recherche-entreprises-mybranch
spec:
  replicas: 1
  selector:
    matchLabels:
      app: recherche-entreprises-front
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/recherche-entreprises
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: recherche-entreprises-front
        application: recherche-entreprises
        component: recherche-entreprises
        owner: recherche-entreprises
        team: recherche-entreprises
        cert: wildcard
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/recherche-entreprises/front:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: recherche-entreprises-front
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 16Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          env:
            - name: REACT_APP_API_URL
              value: >-
                https://search-recherche-entreprises-mybranch.dev2.fabrique.social.gouv.fr/api/v1
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: recherche-entreprises-front
    application: recherche-entreprises
    component: recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  name: recherche-entreprises-front
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/recherche-entreprises
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: recherche-entreprises-mybranch
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: recherche-entreprises-front
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/recherche-entreprises
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: recherche-entreprises-front
    application: recherche-entreprises
    component: recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  name: recherche-entreprises-front
  namespace: recherche-entreprises-mybranch
spec:
  rules:
    - host: recherche-entreprises-mybranch.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: recherche-entreprises-front
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - recherche-entreprises-mybranch.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: netpol-recherche-entreprises-mybranch
  namespace: recherche-entreprises-mybranch
spec:
  ingress:
    - from:
        - podSelector: {}
    - from:
        - namespaceSelector:
            matchLabels:
              network-policy/source: ingress-controller
    - from:
        - namespaceSelector:
            matchLabels:
              network-policy/source: monitoring
  podSelector: {}
  policyTypes:
    - Ingress
"
`;
